arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/fragment/common/CommonFragmentValues.java

```diff
@@ -230,29 +231,78 @@ private long computeGasCostExcludingDeploymentCost() {
     return Hub.GAS_PROJECTOR.of(hub.messageFrame(), hub.opCode()).gasCostExcludingDeploymentCost();
   }
 
-  public long computeGasNext() {
-
-    if (hub.isExceptional()) {
+  /**
+   * Returns the value of the GAS_NEXT column. For CALL's and CREATE's it returns
+   *
+   * <p><center><b>remainingGas - upfrontGasCost</b></center>
+   *
+   * <p>This initial computation has to be amended down the line to account for
+   *
+   * <p>- {@link GasProjection#stipend()} for aborted calls / EOA calls
+   *
+   * <p>- {@link GasProjection#gasPaidOutOfPocket()} when entering a <b>CALL</b>/<b>CREATE</b>
+   *
+   * <p>- precompile specific costs for PRC calls
+   *
+   * <p>The stipend is done through {@link CommonFragmentValues#collectChildStipend(Hub)}}
+   *
+   * @param exceptions
+   * @return
+   */
+  public long computeGasNext(short exceptions) {
+
+    if (Exceptions.any(exceptions)) {
       return 0;
     }
 
     final long gasAfterDeductingCost = computeGasRemaining() - computeGasCost();
 
+    OpCode opCode = hub.opCode();
+    GasProjection gasUtility = Hub.GAS_PROJECTOR.of(hub.messageFrame(), opCode);
+
     return switch (hub.opCodeData().instructionFamily()) {
-      case KEC, COPY, STACK_RAM, STORAGE, LOG, HALT -> (hub.raisesOogxOrIsUnexceptional()
-          ? gasAfterDeductingCost
-          : 0);
-      case CREATE -> gasAfterDeductingCost
-          - Hub.GAS_PROJECTOR.of(hub.messageFrame(), hub.opCode()).gasPaidOutOfPocket();
-      case CALL -> // TODO: this will not work because of 1. aborts with value transfers 2. EOA
-      // calls 3. precompile calls
-      gasAfterDeductingCost
-          - Hub.GAS_PROJECTOR.of(hub.messageFrame(), hub.opCode()).gasPaidOutOfPocket();
+      case KEC, COPY, STACK_RAM, STORAGE, LOG, HALT -> gasAfterDeductingCost;
+      case CREATE -> gasAfterDeductingCost;
+        // TODO: this is only part of the story because of
+        //  1. nonempty init code CREATE's where gas is paid out of pocket
+      case CALL -> gasAfterDeductingCost;
+        // TODO: this is only part of the story because of
+        //  1. aborts with value transfers (immediately reapStipend)
+        //  2. EOA calls with value transfer (immediately reapStipend)
+        //  3. SMC calls: gas paid out of pocket
+        //  4. PRC calls: gas paid out of pocket + special PRC cost + returned gas
+
```

Comment:
This method will have to be refactored later on. Currently it returns the same value in all cases. What changes is how we later on amend it for CALL's and CREATE's.